src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,176 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item, Node<T> next, Node<T> prev) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private void checkIndexAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+    }
+
+    private void checkIndexSet(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+    }
+
+    private void checkIndexRemove(int index) {
+        if (index < 0 || index == size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+    }
+
+    private Node iterateMyLinkedList(int index) {
+        int count = 0;
+        if (index <= size / 2) {
+            Node currentNode = first;
+            while (count != index) {
+                currentNode = currentNode.next;
+                count++;
+            }
+            return currentNode;
+        }
+        if (index > size / 2) {
+            Node currentNode = last;
+            while (count != size - index - 1) {
+                currentNode = currentNode.prev;
+                count++;
+            }
+            return currentNode;
+        }
+        return null;
+    }
+
+    private void unlink(Node nodeToUnlink) {
+        if (nodeToUnlink == first && nodeToUnlink == last) {
+            first = null;
+            last = null;
+            return;
+        }
+        if (nodeToUnlink == first) {
+            nodeToUnlink.next.prev = null;
+            first = nodeToUnlink.next;
+            return;
+        }
+        if (nodeToUnlink == last) {
+            nodeToUnlink.prev.next = null;
+            last = nodeToUnlink.prev;
+            return;
+        }
+        nodeToUnlink.prev.next = nodeToUnlink.next;
+        nodeToUnlink.next.prev = nodeToUnlink.prev;
```

Comment:
```suggestion
        if (nodeToUnlink == first) {
            nodeToUnlink.next.prev = null;
            first = nodeToUnlink.next;
        } else if (nodeToUnlink == last) {
            nodeToUnlink.prev.next = null;
            last = nodeToUnlink.prev;
            return;
        } else {
        nodeToUnlink.prev.next = nodeToUnlink.next;
        nodeToUnlink.next.prev = nodeToUnlink.prev;
       }
```

Reply:
I agree with the change, but have one question. I saw in one tutorial that it is preferred to use multiple if statements instead of if-elseif-else, because "else" case tends to collect various limit conditions (read "mistakes") and generally multiple if statements are easier to read. So, why is it better to use if-elseif-else?

Reply:
to be honest, both options are bad, but choosing between them it's better to use if else in such cases in my opinion. I don't have some specific reasons for using of this approach, just from my experience it's a more readable option than the previous one

Reply:
Understood, thank you!
What do you mean "both options are bad"? How can you avoid if-else constructions at all and in this case in particular?

Reply:
I didn't mean, that you can avoid these options here. I meant that they look both not so good in terms of readability)